solr/core/src/java/org/apache/solr/search/QueryLimits.java

```diff
@@ -16,14 +16,13 @@
  */
 package org.apache.solr.search;
 
-import static org.apache.solr.search.CpuQueryTimeLimit.hasCpuLimit;
-import static org.apache.solr.search.SolrQueryTimeLimit.hasTimeLimit;
+import static org.apache.solr.search.CpuAllowedLimit.hasCpuLimit;
+import static org.apache.solr.search.TimeAllowedLimit.hasTimeLimit;
 
 import java.util.ArrayList;
 import java.util.List;
 import org.apache.lucene.index.QueryTimeout;
 import org.apache.solr.request.SolrQueryRequest;
-import org.apache.solr.util.ThreadCpuTime;
 
 /**
  * Represents the limitations on the query. These limits might be wall clock time, cpu time, memory,
```

Comment:
Maybe this becomes "RequestLimits" or I'm pushing my luck with my suggestions :-)

Reply:
Haha ... it's not a user-facing API, we can always rename it when we want to generalize it.